core/src/test/java/org/apache/iceberg/TestMicroBatchBuilder.java

```diff
@@ -0,0 +1,163 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.util.Collections;
+import java.util.List;
+import org.apache.iceberg.relocated.com.google.common.collect.Iterables;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TestMicroBatchBuilder extends TableTestBase {
+  @Parameterized.Parameters
+  public static Object[][] parameters() {
+    return new Object[][] {
+        new Object[] { 1 },
+        new Object[] { 2 },
+    };
+  }
+
+  public TestMicroBatchBuilder(int formatVersion) {
+    super(formatVersion);
+  }
+
+  @Before
+  public void setupTableProperties() {
+    table.updateProperties().set(TableProperties.MANIFEST_MIN_MERGE_COUNT, "3").commit();
+  }
+
+  @Test
+  public void testGenerateMicroBatch() {
+    add(table.newAppend(), files("A", "B", "C", "D", "E"));
+
+    Snapshots.MicroBatch batch = Snapshots.from(table.snapshot(1L), table.io())
+        .specsById(table.specs())
+        .generate(0, Long.MAX_VALUE, true);
+    Assert.assertEquals(batch.snapshotId(), 1L);
+    Assert.assertEquals(batch.startFileIndex(), 0);
+    Assert.assertEquals(batch.endFileIndex(), 6);
```

Comment:
I think this also needs an assertion for `batch.lastIndexOfSnapshot`

Reply:
Added.